src/workers/search/project-search.js

```diff
@@ -4,32 +4,65 @@
 
 // Maybe reuse file search's functions?
 
+import getMatches from "./get-matches";
+
 export function findSourceMatches(source, queryText) {
-  const { id, loadedState, text } = source;
+  const { loadedState, text } = source;
   if (loadedState != "loaded" || !text || queryText == "") {
     return [];
   }
 
+  const modifiers = {
+    caseSensitive: false,
+    regexMatch: false,
+    wholeWord: false
+  };
+
   const lines = text.split("\n");
-  let result = undefined;
-  const query = new RegExp(queryText, "g");
-
-  const matches = lines
-    .map((_text, line) => {
-      const indices = [];
-
-      while ((result = query.exec(_text))) {
-        indices.push({
-          sourceId: id,
-          line: line + 1,
-          column: result.index,
-          match: result[0],
-          value: _text
-        });
-      }
-      return indices;
-    })
-    .filter(_matches => _matches.length > 0);
-
-  return [].concat(...matches);
+
+  return getMatches(queryText, text, modifiers).map(({ line, ch }) => {
+    const { value, c } = truncateLine(lines[line], ch);
+    return {
+      line: line + 1,
+      column: ch,
+      tc: c,
+      match: queryText,
+      value
+    };
+  });
+}
+
+const startRegex = /([ !@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?])/g;
```

Comment:
could we explain what those regex are about? They seem pretty complex

Reply:
Yeah, probably because they are hand made :P

Start one is to find a word separator near the start of cropped line
And end is to find the same near the end, so the results look nice and clean.